fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveMetaStoreCache.java

```diff
@@ -984,6 +1022,43 @@ public void setAcidInfo(AcidInfo acidInfo) {
         }
     }
 
+    @Data
+    public static class FileSystemCacheKey {
+        private final FileSystemType type;
+        private final JobConf conf;
+        private final long tableId;
+
+        public FileSystemCacheKey(FileSystemType type, JobConf conf, long tableId) {
+            this.type = type;
+            this.conf = conf;
+            this.tableId = tableId;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (this == obj) {
+                return true;
+            }
+            if (!(obj instanceof FileSystemCacheKey)) {
+                return false;
+            }
+            return type.equals(((FileSystemCacheKey) obj).type) && conf == ((FileSystemCacheKey) obj).conf
+                && tableId == ((FileSystemCacheKey) obj).tableId;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(tableId, type);
```

Comment:
Need to take `JobConf` into account. Or if we change the properties, the fs cache is not updated.